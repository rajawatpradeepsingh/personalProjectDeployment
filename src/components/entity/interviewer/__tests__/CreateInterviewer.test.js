// import React from "react";
// import axios from "axios";
// import { render, cleanup, screen, act } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import { CreateInterviewer } from "../CreateInterviewer";
// import  Modal  from "../../Modal";

// jest.mock("axios");
// afterEach(cleanup);
// afterEach(() => jest.resetAllMocks());
// const mockShowModal = jest.fn();
// describe('create interviewer component', () => {

//    //component renders
//    it('renders component', () => {
//       const comp = render(<CreateInterviewer/>);

//       const heading = comp.getByRole("heading");
//       const formInputs = comp.getAllByRole("textbox");
//       const submitBtn = comp.getByTestId("submit-btn");
//       const resetBtn = comp.getByTestId("reset-btn");
//       const closedModalText = screen.queryByText(/interviewer saved successfully!/i);

//       expect(heading).toHaveTextContent("Create Interviewer");
//       expect(formInputs).toHaveLength(9);
//       expect(submitBtn).toBeEnabled();
//       expect(resetBtn).toBeEnabled();
//       expect(closedModalText).not.toBeInTheDocument();
//    });

//    // matches snapshot--update as needed
//    it('matches snapshot DOM node structure', () => {
//       const { asFragment } = render(<CreateInterviewer/>);
//       expect(asFragment()).toMatchSnapshot();
//    });
   
//    //test form initial state
//    it('form loads, fields are empty', () => {
//       const { getByRole, getAllByRole } = render(<CreateInterviewer />);

//       expect(getAllByRole("textbox")).toHaveLength(9);
//       expect(getByRole("textbox", { name: "Last Name *" })).toHaveValue("");
//       expect(getByRole("textbox", { name: "Contact #" })).toHaveValue("");
//       expect(getByRole("textbox", { name: "E-mail *" })).toHaveValue("");
//       expect(getByRole("spinbutton", { name: "Total Experience (in years)" })).toHaveValue(null);
//       expect(getByRole("textbox", { name: "Current Project" })).toHaveValue("");
//       expect(getByRole("textbox", { name: "Interview Skills" })).toHaveValue("");

//    });

//    it('reset button clears form', () => {
//       const { getByRole, getByTestId } = render(<CreateInterviewer />);

//       userEvent.type(getByRole("textbox", {name: "First Name *"}), "Cecilia");
//       userEvent.type(getByRole("textbox", { name: "Last Name *" }), "Rossi");
//       userEvent.type(getByRole("textbox", {name: "Contact #"}), "5102929999");
//       userEvent.type(getByRole("textbox", {name: "E-mail *"}), "test@test.com");
//       userEvent.type(getByRole("spinbutton", {name: "Total Experience (in years)"}), "10");
//       userEvent.type(getByRole("textbox", {name:"Current Project"}), "rcp");
//       userEvent.type(getByRole("textbox", {name:"Interview Skills"}), "test");

//       userEvent.click(getByTestId("reset-btn"));

//       expect(getByRole("textbox", { name: "First Name *" })).toHaveValue("");
//       expect(getByRole("textbox", { name: "Last Name *" })).toHaveValue("");
//       expect(getByRole("textbox", { name: "Contact #" })).toHaveValue("");
//       expect(getByRole("textbox", { name: "E-mail *" })).toHaveValue("");
//       expect(getByRole("spinbutton", {name: "Total Experience (in years)"})).toHaveValue(null);
//       expect(getByRole("textbox", { name: "Current Project" })).toHaveValue("");
//       expect(getByRole("textbox", {name:"Interview Skills"})).toHaveValue("");
      
//    });

//    it("trying to submit without input shows error message", () => {
//       const comp = render(<CreateInterviewer />);

//       const noError = screen.queryByText(/please fill out all required fields/i);
//       expect(noError).not.toBeInTheDocument();

//       userEvent.click(comp.getByTestId("submit-btn"));

//       const errorMessage = comp.getByText(/please fill out all required fields/i);
//       expect(errorMessage).toBeInTheDocument();
//    });

//    it("shows model on successful post", async () => {
//       let modal;
//       const comp = render(<CreateInterviewer />);
//       expect(comp.queryByText(/ok/i)).not.toBeInTheDocument();

//       await act(async() => {
//          await axios.post.mockResolvedValueOnce({status: 201});
//          userEvent.click(comp.getByTestId("submit-btn"));
//          modal = render(<Modal showModal={mockShowModal}></Modal>)
//       })

//       expect(comp.getByText(/ok/i)).toBeInTheDocument();


//    })

// });



