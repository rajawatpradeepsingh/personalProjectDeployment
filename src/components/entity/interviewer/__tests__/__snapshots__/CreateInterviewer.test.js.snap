// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create interviewer component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    id="comp-content"
  >
    <div
      class="comp-header"
    >
      <h1
        data-testid="createInterviewH1"
      >
        Create Interviewer
      </h1>
    </div>
    <div
      class="create-int-error"
    >
      <div
        class="error-message"
      />
    </div>
    <div>
      <form
        enctype="multipart/form-data"
        id="interviewer-form"
      >
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="firstName"
          >
            First Name 
            <span
              class="requiredsign"
            >
              *
            </span>
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="firstName"
            maxlength="20"
            name="firstName"
            pattern="[^0-9]+"
            required=""
            title="Input can't contain numbers"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="lastName"
          >
            Last Name 
            <span
              class="requiredsign"
            >
              *
            </span>
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="lastName"
            maxlength="20"
            name="lastName"
            pattern="[^0-9]+"
            required=""
            title="Input can only contain letters"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="phoneNumber"
          >
            Contact # 
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="phoneNumber"
            maxlength="16"
            name="phoneNumber"
            title="Input can only accept numbers"
            type="tel"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="email"
          >
            E-mail 
            <span
              class="requiredsign"
            >
              *
            </span>
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="email"
            name="email"
            pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,}$"
            required=""
            title="example@example.com"
            type="email"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="city"
          >
            City 
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="city"
            name="city"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="state"
          >
            State / Province 
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="state"
            name="state"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="postalCode"
          >
            ZIP Code 
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="postalCode"
            name="postalCode"
            type="text"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="totalExperience"
          >
            Total Experience 
            <span
              class="guide-text"
            >
              (in years)
            </span>
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="totalExperience"
            min="0"
            name="totalExperience"
            type="number"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <label
            class="input-label"
            for="currentProject"
          >
            Current Project 
          </label>
          <input
            autocomplete="off"
            class="input-field"
            id="currentProject"
            maxlength="30"
            name="currentProject"
            type="text"
            value=""
          />
        </div>
        <div
          class="textarea-container"
        >
          <label
            class="textarea-label"
            for="interviewSkills"
          >
            Interview Skills
          </label>
          <textarea
            autocomplete="off"
            class="textarea"
            id="interviewSkills"
            maxlength="50"
            name="interviewSkills"
          />
        </div>
        <div
          class="create-int-btn-row"
        >
          <input
            class="form-btn"
            data-testid="submit-btn"
            id="int-save-btn"
            name="submit"
            type="submit"
            value="Submit"
          />
          <input
            class="form-btn"
            data-testid="reset-btn"
            id="int-reset-btn"
            name="cancel"
            type="reset"
            value="Reset"
          />
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
